extensions/amp-imgur/0.1/amp-imgur.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /**
+  * @fileoverview Embeds a imgur
+  * Example:
+  * <code>
+  * <amp-imgur
+  *   layout="reponsive"
+  *   width="540"
+  *   height="663"
+  *   data-imgur-id="f462IUj">
+  * </amp-imgur>
+  * </code>
+  */
+
+import {user} from '../../../src/log';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {tryParseJson} from '../../../src/json';
+import {isObject} from '../../../src/types';
+import {listen} from '../../../src/event-helper';
+
+export class AmpImgur extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
```

Comment:
no need to keep a reference to the promise. just return `this.loadPromise(iframe);` in `layoutCallback`